From 90e5050456387317c70dd8a3668940e0d9ab61f2 Mon Sep 17 00:00:00 2001
From: Jerry Jelinek <jerry.jelinek@joyent.com>
Date: Wed, 15 May 2019 17:04:54 +0000
Subject: [PATCH] OS-7798 zfs_zone_zio_dequeue: count==0 console noise during
 boot Reviewed by: Kody Kantor <kody.kantor@joyent.com> Approved by: Kody
 Kantor <kody.kantor@joyent.com>

---
 usr/src/uts/common/fs/zfs/vdev_queue.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/usr/src/uts/common/fs/zfs/vdev_queue.c b/usr/src/uts/common/fs/zfs/vdev_queue.c
index 3c9057b943..f76600f8e5 100644
--- a/usr/src/uts/common/fs/zfs/vdev_queue.c
+++ b/usr/src/uts/common/fs/zfs/vdev_queue.c
@@ -21,7 +21,7 @@
 /*
  * Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
  * Use is subject to license terms.
- * Copyright 2013, Joyent, Inc. All rights reserved.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -864,9 +864,11 @@ vdev_queue_change_io_priority(zio_t *zio, zio_priority_t priority)
 		spa_t *spa = zio->io_spa;
 		zio_priority_t oldpri = zio->io_priority;
 
+		zfs_zone_zio_dequeue(zio);
 		avl_remove(vdev_queue_class_tree(vq, zio->io_priority), zio);
 		zio->io_priority = priority;
 		avl_add(vdev_queue_class_tree(vq, zio->io_priority), zio);
+		zfs_zone_zio_enqueue(zio);
 
 		mutex_enter(&spa->spa_iokstat_lock);
 		ASSERT3U(spa->spa_queue_stats[oldpri].spa_queued, >, 0);
-- 
2.21.0

